addons/sale/models/sale_order.py

```diff
@@ -72,24 +72,18 @@ def _rec_names_search(self):
         states=READONLY_FIELD_STATES,
         domain="[('type', '!=', 'private'), ('company_id', 'in', (False, company_id))]")
     state = fields.Selection(
-        selection=[
-            ('draft', "Quotation"),
-            ('sent', "Quotation Sent"),
-            ('sale', "Sales Order"),
-            ('done', "Locked"),
-            ('cancel', "Cancelled"),
-        ],
+        selection=SALE_ORDER_STATE,
         string="Status",
         readonly=True, copy=False, index=True,
         tracking=3,
         default='draft')
+    locked = fields.Boolean(default=False, copy=False)
```

Comment:
Thinking of it, the comportment of this field can be confusing sometime, so maybe add a description to explain that if locked, it can be changed (obvisouly) but also that some logic might be blocked too? (I.e. recomputation of taxes, stock rules, etc. ?)